File: frontend/src/objects/merchant/apiTypes/UpdateStoreOperationalRequest.ts
// Business note: service protocol DTO shared with backend apiTypes; keep field names and value object types aligned.
import type { BusinessHours } from '@/objects/merchant/store/BusinessHours'
import type { AddressText, DeliveryCoordinate, Minutes } from '@/objects/core/SharedObjects'

export type UpdateStoreOperationalRequest = {
  storeAddress: AddressText
  location?: DeliveryCoordinate
  businessHours: BusinessHours
  avgPrepMinutes: Minutes
}

File: backend/src/main/scala/services/merchant/objects/apiTypes/UpdateStoreOperationalRequest.scala
package services.merchant.objects.apiTypes

// Business note: protocol DTO shared with the frontend; keep field names and value object types mirrored across stacks.
import services.merchant.objects.*

import system.objects.given
import services.customer.objects.*

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class UpdateStoreOperationalRequest(
    storeAddress: AddressText,
    location: Option[StoreLocation],
    businessHours: BusinessHours,
    avgPrepMinutes: Minutes,
)
object UpdateStoreOperationalRequest:
  given Encoder[UpdateStoreOperationalRequest] = deriveEncoder
  given Decoder[UpdateStoreOperationalRequest] = deriveDecoder
